### PR TITLE
[Cell Registration] auto register table view cells

### PR DESCRIPTION
**Pull request by jessesquires**
- Refactor `ViewRegistrationMethod` case names
- Introduce `ViewRegistrationInfo` to encapsulate registration data
- Use the class name as the default cell identifier

#18

### CODE REVIEW

**Comment by jessesquires** [comment]
Next up: doing the same for collection view

File: Example/ToolTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//  License
+//  Copyright © 2018-present PlanGrid, Inc.
+//  Released under an MIT license: https://opensource.org/licenses/MIT
+// 
+
+import UIKit
+
+final class ToolTableViewCell: UITableViewCell {
+
+}
```
Comment (jessesquires) [inline Example/ToolTableViewCell.swift]: one "opinion" encoded in the model:

we get rid of the idea of "reuseIds" and use the class name for the cell. this means you'll need unique cell class names.

however, i think think we should encourage not just using `UITableViewCell`

**Comment by PlanBot-iOS** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Source code changes (in APIs or behaviors) should have an entry in CHANGELOG.md.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Example/TableViewCellModels.swift [hunk 2]
```diff
@@ -18,11 +18,11 @@ import Foundation
 import ReactiveLists
 
 struct ToolTableCellModel: TableViewCellViewModel, DiffableViewModel {
-    var accessibilityFormat: CellAccessibilityFormat = ""
-    let cellIdentifier = "TableUserCell"
+    let registrationInfo = ViewRegistrationInfo(classType: ToolTableViewCell.self)
```
Comment (anayini) [inline Example/TableViewCellModels.swift]: 😍

**Pull request merged**